### DISCUSSION TITLE
TI99/4a

### DISCUSSION THREAD

**Discussion by Seawolf409** [keirf/flashfloppy#873] [Q&A]
I have well over 100 disks for the TI99/4a and would like to be able to read these to my Windows PC.  Ideally I would like Windows/File  Explorer to be able to show that the files are there on the PC, with the files in either text or binary as appropriate. However anything that gives me access to the files is fine.  I have no idea how to do this with Greaseweazel, but can read the floppies OK.
I have read 3 of these disks and the raw output is here:
https://blunham.com/for/Github
The only disk I know to be DSDD is named as such, the others could be SSSD, particularly the EditorAssembler.
The format of the disks is described in a TI document:
https://blunham.com/Misc/Texas
in the "Software Specification for Disk Peripheral", section 6.
As you can imagine, any help with this would be much appreciated.
Regards, Colin.

**Comment by cas206** [comment]
This is the wrong project for this discussion.  Here is the maintainers GreezeWeazle prorject.
https://github.com/keirf/greaseweazle/discussions

However, you can start here getting some ideas at the following links.
https://forums.atariage.com/topic/353091-grease-weazle-and-ti-disk-images/
https://www.youtube.com/watch?v=3-Ut-Big0bQ

**Comment by keirf** [comment]
Some more advice:

1. You can load these raw files into HxC software (https://hxc2001.com/download/floppy_drive_emulator/) see the link "HxC Floppy Emulator software (beta/snapshot version)". There are prebuilts for Windows and Mac. For Linux you can build from source https://github.com/jfdelnero/HxCFloppyEmulator

2. Within HxC software you can inspect your raw images using the "Track Analyzer" button. It's very useful once you know your way round it. Here we can see your raw images look good, but that they are 40-cyl disks dumped in a 80-cyl drive. You could get rid of the extra tracks something like: `gw convert --tracks=c=0-39:step=2 in/00.0.raw out/00.0.raw`

3. The Export button will allow you to export the raw images to "V9T9 DSK" format. I think this probably works even if you don't strip out the empty alternating tracks!

4. There are likely existing tools to inspect V9T9 DSK files. For example VDM99 on Windows. I found this with some cursory Googling -- I'm sure there are other tools too!

Ultimately Greaseweazle should (will) support V9T9 DSK image files directly, but it's not there yet. Nonetheless there are other tools that will get you there, it just makes the tool pipeline a bit longer.

**Comment by Seawolf409** [comment]
Hi Keir,
Thank you very much for your help.  I can see the data on the disks now, but it's a mess for reasons I don't understand.  I'm going to have to find a floppy with a file on it for which I have a copy for the PC and then compare the two so I can write a bit of software to fix the problem.  As these are on Magneto Optical disks, I'm going to have to set up SCSI on this machine - could take a while !

I'll come back to you if I need help - I'll post to the greaseweazle/discussions group this time !

Best wishes,
Colin.
